overlord/ifacestate/hotplug/udevmon.go

```diff
@@ -139,3 +141,16 @@ func (m *UDevMonitor) removeDevice(kobj string, env map[string]string) {
 		m.deviceRemovedCb(di)
 	}
 }
+
+func MockCreateUDevMonitor(new func(DeviceAddedCallback, DeviceRemovedCallback) UDevMon) (restore func()) {
```

Comment:
I would suggest making a type for the function so that it reads more easily:

```
type newMonitorFn func(DeviceAddedCallback, DeviceRemovedCallback) UDevMon
```